docs/source/llms/llm-evaluate/notebooks/huggingface-evaluation.ipynb

```diff
@@ -0,0 +1,1099 @@
+       "6  The fraction 4/16 is equivalent to 1/4 because...   \n",
+       "7  John was at a crossroads in his life. He had j...   \n",
+       "8  <nooutput> This type of instruction cannot be ...   \n",
+       "9      He finished his meal and left the restaurant.   \n",
+       "\n",
+       "                                                text  \n",
+       "0  Below is an instruction that describes a task....  \n",
+       "1  Below is an instruction that describes a task....  \n",
+       "2  Below is an instruction that describes a task....  \n",
+       "3  Below is an instruction that describes a task....  \n",
+       "4  Below is an instruction that describes a task....  \n",
+       "5  Below is an instruction that describes a task,...  \n",
+       "6  Below is an instruction that describes a task,...  \n",
+       "7  Below is an instruction that describes a task....  \n",
+       "8  Below is an instruction that describes a task....  \n",
+       "9  Below is an instruction that describes a task,...  "
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset = load_dataset(\"tatsu-lab/alpaca\")\n",
+    "eval_df = pd.DataFrame(dataset[\"train\"])\n",
+    "eval_df.head(10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Metrics"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we are evaluating how well our model can provide an answer to a given instruction, we may want to choose some metrics to help measure this on top of any builtin metrics that `mlflow.evaluate()` gives us.\n",
+    "\n",
+    "Let's measure how well out model is doing on following two metrics:\n",
+    "- Is the answer correct? Let's use the predefined metric `answer_correctness` here.\n",
+    "- Is the answer fluent, clear, and concise? We will define a custom metric `answer_quality` to measure this.\n",
+    "\n",
+    "We will need to pass these into `extra_metrics` argument for `mlflow.evaluate()`.\n",
+    "\n",
+    "#### What is an Evaluation Metric?\n",
+    "\n",
+    "An evaluation metric encapsulates any quantitative or qualitative measure you want to calculate for your model. For each model type, `mlflow.evaluate()` will automatically calculate some set of builtin metrics. Refer [here](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate) for which builtin metrics will be calculated for each model type. You can also pass in any other metrics you want to calculate as extra metrics. MLflow provides a set of predefined metrics that you can find [here](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html), or you can define your own custom metrics. In the example here, we will use the predefined metric `mlflow.metrics.genai.answer_correctness`.\n",
+    "\n",
+    "##### Custom Metrics\n",
```

Comment:
Can we skip describing `make_metric` explanation? Doesn't seem to be related to what we are doing here.